scripts/west_commands/sbom/input_build.py

```diff
@@ -427,9 +434,16 @@ def generate_input(data: Data):
         log.wrn('Fetching input files from a build directory is experimental for now.')
         check_external_tools(Path(args.build_dir[0][0]))
         for build_dir, *targets in args.build_dir:
-            if len(targets) == 0:
-                targets = [DEFAULT_TARGET]
-            log.dbg(f'INPUT: build directory: {build_dir}, targets: {targets}')
-            b = InputBuild(data, build_dir)
-            for target in targets:
-                b.generate_from_target(target)
+            try:
+                domains = yaml.safe_load(open(os.path.join(build_dir, 'domains.yaml'), 'r'))
+                domains = [d['name'] for d in domains['domains']]
+            except:
+                domains = ['.']
+            for domain in domains:
+                domain_build_dir = Path(os.path.join(build_dir, domain))
+                if len(targets) == 0:
+                    targets = [DEFAULT_TARGET]
+                log.dbg(f'INPUT: build directory: {domain_build_dir}, targets: {targets}')
+                b = InputBuild(data, domain_build_dir)
+                for target in targets:
+                    b.generate_from_target(target)
```

Comment:
In the future, we can consider separating report based on domains, but for now it is ok to put them into one bucket.